### PR TITLE
Fix exit, adding debug messages and cleaning

### LINKED ISSUE #231: Game is often not ended properly
While this problem does not seem to be faced during automated tests, when running the games on AWS, it often ends up that `webots_bin` is still running after the game was terminated with a message that sounds sane (e.g.):

```
INFO: HTML5 animation successfully exported in '/home/ubuntu/webots/projects/samples/contests/robocup/controllers/referee/TEST48.html'

Interrupting game automatically after 180 seconds
Game interrupted before the end.
Start listening to AutoReferee connection on port 8750
INFO: referee: Terminating.
INFO: player: Terminating.
```

There's probably something wrong in the order the process are stopped but with current verbosity it's pretty hard to understand what exactly is failing.

### LINKED ISSUE #233: More detailed logs
With respect to #229 #231 and #232 the debugging is pretty difficult because we are lacking information, therefore we should add more message to the referee log:

- [x] For all the different phases of `clean_up`
- [x] Each time we are setting up the `ready_real_time` or `set_real_time` `ready_countdown` `interruption_countdown` `set_countdown`

### PR DESCRIPTION
**Pull request by medrimonia**
Fixes #233.
Fixes #231.

Fix exit:
- There was an infinite loop in `clean_exit` when game.over was True

Cleaning:
- Removed the `set_countdown` variable (initialized to 0 and never changed)

Debug messages:
- Adding many of them on initialization of countdown values to make debugging
  easier

### CODE REVIEW

**Comment by medrimonia** [comment]
I added an unrelated element which removes the `sent_finish` variable that does not seem useful anymore.

All tests are passing with this new change while they were previously not.

File: projects/samples/contests/robocup/controllers/referee/referee.py [hunk 1]
```diff
@@ -2748,13 +2760,13 @@ def read_team(json_path):
...
                     game_controller_send('STATE:READY')
-            elif not game.state.first_half and game.sent_finish:
-                game.sent_finish = False
+            elif not game.state.first_half:
                 game_type = ''
                 if game.overtime:
                     game_type = 'overtime '
                 info(f'Beginning of {game_type}second half.')
                 kickoff()
+                info(f'Going to READY in {HALF_TIME_BREAK_REAL_TIME_DURATION}')
```
Comment (omichel) [inline projects/samples/contests/robocup/controllers/referee/referee.py]:
```suggestion
                info(f'Going to READY in {HALF_TIME_BREAK_REAL_TIME_DURATION} seconds (real-time)')
```

File: projects/samples/contests/robocup/controllers/referee/referee.py [hunk 2]
```diff
@@ -2712,22 +2720,24 @@ def read_team(json_path):
...
                     info('Beginning of the knockout first half.')
                     game_controller_send('STATE:OVERTIME-FIRST-HALF')
+                    info(f'Going to READY in {HALF_TIME_BREAK_REAL_TIME_DURATION}')
                     game.ready_real_time = time.time() + HALF_TIME_BREAK_REAL_TIME_DURATION
             elif game.type == 'KNOCKOUT' and game.state.teams[0].score == game.state.teams[1].score:
                 if game.ready_real_Time is None:
                     info('Beginning of penalty shout-out.')
                     game_controller_send('STATE:PENALTY-SHOOTOUT')
                     game.penalty_shootout = True
+                    info(f'Going to READY in {HALF_TIME_BREAK_REAL_TIME_DURATION}')
```
Comment (omichel) [inline projects/samples/contests/robocup/controllers/referee/referee.py]:
```suggestion
                    info(f'Going to READY in {HALF_TIME_BREAK_REAL_TIME_DURATION} seconds (real-time)')
```

File: projects/samples/contests/robocup/controllers/referee/referee.py [hunk 3]
```diff
@@ -2712,22 +2720,24 @@ def read_team(json_path):
...
                 if game.state.seconds_remaining <= 0:
                     next_penalty_shootout()
             elif game.state.first_half:
+                info("Received state FINISHED: end of first half")
                 game.ready_real_time = None
             elif game.type == 'KNOCKOUT' and game.overtime and game.state.teams[0].score == game.state.teams[1].score:
                 if game.ready_real_time is None:
                     info('Beginning of the knockout first half.')
                     game_controller_send('STATE:OVERTIME-FIRST-HALF')
+                    info(f'Going to READY in {HALF_TIME_BREAK_REAL_TIME_DURATION}')
```
Comment (omichel) [inline projects/samples/contests/robocup/controllers/referee/referee.py]:
```suggestion
                    info(f'Going to READY in {HALF_TIME_BREAK_REAL_TIME_DURATION} seconds (real-time)')
```

File: projects/samples/contests/robocup/controllers/referee/referee.py [hunk 4]
```diff
@@ -2437,9 +2442,11 @@ def read_team(json_path):
         info(f'{"Red" if game.kickoff == game.red.id else "Blue"} team will start the penalty shoot-out.')
         game.phase = 'PENALTY-SHOOTOUT'
         game.ready_real_time = None
-        game.set_real_time = time.time() + REAL_TIME_BEFORE_FIRST_READY_STATE  # real time for ready state (initial kick-off)
+        info(f'Penalty start: Waiting {REAL_TIME_BEFORE_FIRST_READY_STATE} seconds before going to SET')
+        game.set_real_time = time.time() + REAL_TIME_BEFORE_FIRST_READY_STATE  # real time for set state (penalty-shootout)
         game_controller_send(f'KICKOFF:{game.kickoff}')
     else:
+        info(f'Regular start: Waiting {REAL_TIME_BEFORE_FIRST_READY_STATE} seconds before going to READY')
```
Comment (omichel) [inline projects/samples/contests/robocup/controllers/referee/referee.py]:
```suggestion
        info(f'Regular start: Waiting {REAL_TIME_BEFORE_FIRST_READY_STATE} seconds (real-time) before going to READY')
```

File: projects/samples/contests/robocup/controllers/referee/referee.py [hunk 5]
```diff
@@ -2437,9 +2442,11 @@ def read_team(json_path):
         info(f'{"Red" if game.kickoff == game.red.id else "Blue"} team will start the penalty shoot-out.')
         game.phase = 'PENALTY-SHOOTOUT'
         game.ready_real_time = None
-        game.set_real_time = time.time() + REAL_TIME_BEFORE_FIRST_READY_STATE  # real time for ready state (initial kick-off)
+        info(f'Penalty start: Waiting {REAL_TIME_BEFORE_FIRST_READY_STATE} seconds before going to SET')
```
Comment (omichel) [inline projects/samples/contests/robocup/controllers/referee/referee.py]:
```suggestion
        info(f'Penalty start: Waiting {REAL_TIME_BEFORE_FIRST_READY_STATE} seconds (real-time) before going to SET')
```

File: projects/samples/contests/robocup/controllers/referee/referee.py [hunk 6]
```diff
@@ -143,8 +148,10 @@ def clean_exit():
                     break
         else:
             waiting_steps = END_OF_GAME_TIMEOUT * 1000 / time_step
+            info(f"Waiting {waiting_steps} before exiting")
```
Comment (omichel) [inline projects/samples/contests/robocup/controllers/referee/referee.py]:
```suggestion
            info(f"Waiting {waiting_steps} simulation steps before exiting")
```

**Review by omichel — APPROVED** [review]
I have some suggestions to make the logs a bit more precise.

**Pull request merged**